parcels/particleset.py

```diff
@@ -249,13 +249,14 @@ def execute(self, pyfunc=AdvectionRK4, starttime=None, endtime=None, dt=1.,
         assert(timeleaps >= 0)
         leaptime = starttime
         for _ in range(timeleaps):
-            leaptime += interval
-            self.kernel.execute(self, endtime=leaptime, dt=dt,
-                                recovery=recovery)
+            # First write output_file, because particles could have been added
```

Comment:
The problem if we do another `write` at the end, is that if we then restart execution (say in a loop), we output twice. The looping means it's a choice whether we output initial or final positions.

Another option is to add a test to ParticleFile.write to check if a certain time has been written already. Then we can write both at the start and the end of the `.execute`. I'll have a look into that